chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/conformancemethodconversion/ConformanceMethodConversionJob.java

```diff
@@ -0,0 +1,287 @@
+                    }
+                }
+            });
+        } else {
+            LOGGER.info("Could not run job - 'conformance-method' flag is not on.");
+        }
+        LOGGER.info("********* Completed the Conformance Method Conversion job. *********");
+    }
+
+    private void convertListing(CertifiedProductSearchDetails listing) {
+        listing.getCertificationResults().stream()
+                .filter(cr -> cr.isSuccess())
+                .forEach(cr -> convertCertificationResult(listing, cr));
+    }
+
+    private void convertCertificationResult(CertifiedProductSearchDetails listing, CertificationResult result) {
+        List<ConversionRule> applicableConversionRules = conversionRules.stream()
+                .filter(rule -> rule.getCriterion().equals(result.getCriterion()))
+                .toList();
+
+        applicableConversionRules.stream()
+                .forEach(rule -> result.getTestProcedures().stream()
+                        .forEach(crtp -> convertTestProcedureToConformanceMethod(listing, result, crtp, rule)));
+    }
+
+    private void convertTestProcedureToConformanceMethod(CertifiedProductSearchDetails listing, CertificationResult cr,  CertificationResultTestProcedure crtp, ConversionRule rule) {
+        if (doesTestProcedureMatchRule(crtp, cr.isGap(), rule)) {
+            deleteCertificationResultTestProcedure(crtp);
+            addCertificationResultConformanceMethod(cr, rule.getConformanceMethodName(), crtp.getTestProcedureVersion());
+            LOGGER.info(listing.getChplProductNumber() + " | "
+                    + String.format("Criterion: %s - Convert TP(%s) to CM(%s)",
+                            CertificationCriterionService.formatCriteriaNumber(cr.getCriterion()),
+                            crtp.getTestProcedure().getName(),
+                            rule.getConformanceMethodName()));
+        }
+    }
+
+
+    private void deleteCertificationResultTestProcedure(CertificationResultTestProcedure crtp) {
+        conformanceMethodConversionDAO.deleteCertificationResultTestProcedure(crtp.getId());
+    }
+
+    private void addCertificationResultConformanceMethod(CertificationResult cr, String conformanceMethodName, String version) {
+        conformanceMethodConversionDAO.addCertificationResultConformanceMethod(cr.getId(), conformanceMethodName, version);
+    }
+
+    private Boolean doesTestProcedureMatchRule(CertificationResultTestProcedure crtp, Boolean hasGap, ConversionRule rule) {
+        Boolean gapCheck = rule.getHasGap() == null ? true : hasGap == rule.getHasGap();
+        return (StringUtils.isNotEmpty(crtp.getTestProcedure().getName())
+                        && rule.getTestProcedureName().equals(WILDCARD)
+                        && gapCheck)
+                || (StringUtils.isNotEmpty(crtp.getTestProcedure().getName())
+                        && rule.getTestProcedureName().equals(crtp.getTestProcedure().getName())
+                        && gapCheck);
```

Comment:
I feel like this statement is really hard to parse. Both of the main clauses on each side of the `||` have two of the three sub-clauses that are identical. It's like

```
if (a & b1 & c 
|| a & b2 & c)
```

Would it be the same logic / read more clearly if it were instead something like
```
if (a & c & (b1 | b2))
```